sklearn/neighbors/tests/test_nearest_centroid.py

```diff
@@ -147,3 +148,37 @@ def test_manhattan_metric():
     clf.fit(X_csr, y)
     assert_array_equal(clf.centroids_, dense_centroid)
     assert_array_equal(dense_centroid, [[-1, -1], [1, 1]])
+
+
+def test_partial_fit():
+    # Test the partial fitting
+
+    clf = NearestCentroid()
+    clf.partial_fit(X[:3], y[:3], classes=[-1, 1])
+    clf.partial_fit(X[3:], y[3:])
+    assert_array_equal(clf.predict(T), true_result)
+
+    X2 = [[-2, -1], [-1, -2], [1, 1], [-1, -1], [1, 2], [2, 1]]
+    y2 = [-1, -1, 1, -1, 1, 1]
+    clf = NearestCentroid()
+    clf.partial_fit(X2[:3], y2[:3], classes=[-1, 1])
+    assert_array_equal(clf.predict(T), [-1, 1, 1])
+    clf.partial_fit(X2[3:], y2[3:])
+    assert_array_equal(clf.predict(T), true_result)
+
+
+def test_partial_shrinkage_correct():
+    # Ensure that the shrinking is correct.
+    # The expected result is calculated by R (pamr),
+    # which is implemented by the author of the original paper.
+    # (One need to modify the code to output the new centroid in pamr.predict)
```

Comment:
Modify which code? I don't understand this. You could just show the R code used to derive these numbers.